app/main.py

```diff
@@ -1,8 +1,49 @@
+from typing import List, Dict, Union, Optional
+
+
 class Person:
-    # write your code here
-    pass
+    # Class-level dictionary to store all Person instances by name
+    people: Dict[str, "Person"] = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        """
+        Initialize a Person instance with a name and age.
+        Add the instance to the class-level people dictionary.
+        """
+        self.name: str = name
+        self.age: int = age
+        # Register the Person instance in the class attribute people
+        Person.people[name] = self
+
+    def __setattr__(self, key: str, value: Optional["Person"]) -> None:
+        """
+        Override __setattr__ to dynamically add wife/husband attributes
+        only if they are set outside the constructor.
+        """
+        if key in {"wife", "husband"} and value is not None:
+            super().__setattr__(key, value)
+        else:
+            super().__setattr__(key, value)
+
+
+def create_person_list(people: List[Dict[str, Union[(
+        str, int, None)]]]) -> List[Person]:
```

Comment:
The type hint for the `people` parameter is incorrect. The `Union` type should not have square brackets inside it. It should be `Union[str, int, None]` instead of `Union[(str, int, None)]`.